src/main/scala/com/amazon/deequ/schema/ColumnName.scala

```diff
@@ -0,0 +1,110 @@
+/**
+  * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+  * use this file except in compliance with the License. A copy of the License
+  * is located at
+  *
+  *     http://aws.amazon.com/apache2.0/
+  *
+  * or in the "license" file accompanying this file. This file is distributed on
+  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+  * express or implied. See the License for the specific language governing
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.schema
+
+object ColumnName {
+
+  type Sanitized = Either[SanitizeError, String]
+
+  /**
+    * Sanitizes the input column name by ensuring that it is escaped with backticks.
+    *
+    * The resulting String is the escaped input column name, which is safe to use in
+    * any Spark SQL statement.
+    */
+  def sanitizeForSql(columnName: String): Sanitized =
+    if (columnName == null) {
+      Left(NullColumn)
+
+    } else {
+      val (prefix, suffix, rawColumnName) = {
+        val prefix = if (!columnName.startsWith("`")) "`" else ""
```

Comment:
I think this can be simplified, because columnName.startsWith("`") in this line and prefix.isEmpty later basically check the same condition

Reply:
It's also used in creating the sanitized string (in the final `if-else` expression: the `Right(s"$prefix$rawColumnName$suffix")` expression.

Reply:
Pushed a change that I think simplifies this all:
- raw column name extraction is a helper function
- `startsWith` & `endsWith` checks are in helper functions

```scala
      val rawColumnName = extractColumnName(columnName)
      if (rawColumnName.contains("`")) {
        Left(ColumnNameHasBackticks(columnName))
      } else {
        val prefix = if (hasStartingTick(columnName)) "" else "`"
        val suffix = if (hasEndingTick(columnName)) "" else "`"
        Right(s"$prefix$rawColumnName$suffix")
      }
```

Reply:
```scala

  /** Extracts the original column name from a possibly sanitized column name. */
  @inline
  private[this] def extractColumnName(name: String): String =
    name.slice(
      if (hasStartingTick(name)) 1 else 0,
      name.length - (if (hasEndingTick(name)) 1 else 0)
    )

  /** True iff the input string starts with a backtick. False otherwise. */
  @inline
  private[this] def hasStartingTick(s: String): Boolean =
    s.startsWith("`")

  /** True iff the input string ends with a backtick. False otherwise. */
  @inline
  private[this] def hasEndingTick(s: String): Boolean =
    s.endsWith("`")
```